config/initializers/pagelets.rb

```diff
@@ -0,0 +1,6 @@
+# initialize katello related pagelets
+mgr = Pagelets::Manager.new "hosts/_form"
+
+mgr.add_pagelet :main_tab_fields,
+  :partial => "overrides/activation_keys/host_environment_select",
+  :priority => 200
```

Comment:
this places the lifecycle end environment selector below the puppet environment selector, but I think it should be above, thoughts on lowering the priority?